libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,559 @@
+    ElementRef, AfterContentInit, Output, EventEmitter,
+    HostListener, ChangeDetectorRef, OnInit, TemplateRef, AfterViewInit, OnDestroy, forwardRef, SimpleChanges, ContentChild
+} from '@angular/core';
+import { BaseComponent } from '../base';
+import { FocusKeyManager } from '@angular/cdk/a11y';
+import { UP_ARROW, DOWN_ARROW, ENTER, SPACE } from '@angular/cdk/keycodes';
+import { BaseListItem, ListItemDef } from './base-list-item';
+import { SelectionModel } from '@angular/cdk/collections';
+import { ListDataSource, isDataSource } from '../../domain/data-source';
+import { Subscription, Subject, of } from 'rxjs';
+import { takeUntil, delay, tap } from 'rxjs/operators';
+import { ContentDensity } from '../../components/form/form-control';
+import { ListConfig } from './list.config';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+
+
+export type SelectionType = '' | 'multi' | 'single' | 'delete';
+export type ListType = 'inactive' | 'active' | 'detail';
+let nextListId = 0;
+let nextListGrpHeaderId = 0;
+export type FdpListDataSource<T> = ListDataSource<T> | T[];
+export const LIST_VALUE_ACCESSOR: any = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => ListComponent),
+    multi: true
+};
+
+
+/**
+ * The List component represents a container for list item types.
+ * It is used to display a list features.
+ */
+@Component({
+    selector: 'fdp-list',
+    templateUrl: './list.component.html',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./list.component.scss',
+        './drag-and-drop.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [LIST_VALUE_ACCESSOR],
+})
+
+
+export class ListComponent extends BaseComponent implements AfterContentInit, OnInit, AfterViewInit, OnDestroy, ControlValueAccessor {
+
+
+    /**
+    * Child items of the List.
+    */
+    @ContentChildren(BaseListItem, { descendants: true }) ListItems: QueryList<BaseListItem>;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    keyManager: FocusKeyManager<BaseListItem>;
+
+    /** @hidden */
+    @Output()
+    selectedItemChange: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Whether Navigation mode is included to list component
+     * for all the items
+    */
+    _navigated: boolean;
+
+    /** Whether Navigation mode is included to list component
+     * only a subset of the list items are navigable
+     * you should indicate those by displaying a navigation arrow
+    */
+    _navigationIndicator: boolean;
+
+    /** Whether By line is present in list item*/
+    _hasByLine: boolean;
+
+    /**Enables lazy loadMore of data */
+    @Input()
+    loadMore: boolean;
+
+    /**To display loading symbol */
+    loading = false;
+
+    /**Enables data load on scroll for true
+     * false: enables data loading on button
+     * click
+     */
+    @Input()
+    onScroll: boolean;
```

Comment:
Incorrect name for boolean property. It looks like event

Reply:
Renamed to loadOnScroll.